Multiphysics boundary conditions migration for HT and Tracer (#359)
lethe-cfd/lethe: doc/source/parameters/cfd/boundary_conditions_multiphysics.rst

==================================
Multiphysics - Boundary Contitions
==================================

This subsection's purpose is defining the boundary conditions associated to multiphysic problems. 

> Heat Transfer
> ^^^^^^^^^^^^^
> 
> .. code-block:: text
> 

Reviewer: ```suggestion
Heat Transfer
^^^^^^^^^^^^^

For heat transfer boundary conditions, the defaults parameters are:

.. code-block:: text

```

    subsection boundary conditions heat transfer
    set number                  = 2
        subsection bc 0
            set type	        = temperature
            set value	        = 0
        end

>         subsection bc 1

Reviewer: I think it is a good idea to devide two boundary conditions for two possibilities! Only thing, to standardize with the other documentation, we could write the default parameters. They are 0 both Tinf and h.

            set type		    = convection
            set h 		    = 10
            set Tinf	   	    = -13
        end
    end

* ``number``: This is the number of boundary conditions of the problem. 

* ``type`` : This is the type of boundary condition been imposed. At the moment, choices are 
* ``temperature`` (Dirichlet BC), to impose a given temperature ``value`` at the boundary 

> * ``convection`` (Robin BC) for cooling/heating, depending on the environment temperature at the boundary ``Tinf``, with a given heat transfer coefficient ``h`` following Newton's law of cooling (and heating)

Reviewer: I'm not able to do it on the browser, but you could insert a tab in front of the temperature and convection bullets, it will do as in https://lethe-cfd.github.io/lethe/parameters/cfd/physical_properties.html

Author: You mean as a second level for the bullet points?

Reviewer: yes! :)

.. math::
    \frac{ \partial T}{\partial \mathbf{n}} = h (T - T_\textit{inf})

.. note::
    By default, the boundary condition type is ``temperature`` with ``value = 0``

> 
> Tracer
> ^^^^^^
> .. code-block:: text
> 
>     subsection boundary conditions tracer

Reviewer: ```suggestion

Tracer
^^^^^^

For tracer boundary conditions, the defaults parameters are:

.. code-block:: text

    subsection boundary conditions tracer
```

    set number                  = 1
        subsection bc 0
                set type              = dirichlet
                subsection dirichlet

>                     set Function expression = x

Reviewer: The default function is 0, maybe we could add this.

==================================
Multiphysics - Boundary Contitions
==================================

This subsection's purpose is defining the boundary conditions associated to multiphysic problems. 

Heat Transfer
^^^^^^^^^^^^^

> For heat transfer boundary conditions, the defaults parameters are:

Reviewer 2: ```suggestion
For heat transfer boundary conditions, possible parameters are:
```

(the default value is only temperature = 0, no convection)

Reviewer: the type parameter is set to temperature by default, but when using a convection type, these are the default parameters. Maybe we could rephrase that

Author: I rephrased it, but the branch was merged before I finished. I'll put it in the next PR as a bonus

.. code-block:: text

    subsection boundary conditions heat transfer
    set number                  = 2
        subsection bc 0
            set type	        = temperature
            set value	        = 0
        end
        subsection bc 1
            set type		    = convection
            set h 		    = 0
            set Tinf	   	    = 0
        end
    end

* ``number``: This is the number of boundary conditions of the problem. 

> 

Reviewer 2: ```suggestion
.. warning::
    The number of boundary conditions must be specified explicitly as the ParameterHandler is not able to deduce the number of boundary conditions from the number of ``bc`` subsections. This is often a source of error.
```

I think it never hurts to say it again here ;)
